File: sort_random_array.js
const MAX_ELEMENTS = 1000000;
const MAX_VALUE = 10000000;

const arr = [];

console.time("arrayGeneration");
for (let at = 0; at < MAX_ELEMENTS; ++at) {
    arr[at] = Math.floor(Math.random() * MAX_VALUE);
}
console.timeEnd("arrayGeneration");

console.time("arraySort");
arr.sort((a, b) => a - b);
console.timeEnd("arraySort");

File: sort_random_array.cpp
#include <cstdlib>
#include <ctime>
#include <ctime>
#include <cstdint>

#include <algorithm>
#include <chrono>
#include <vector>
#include <iostream>

using namespace std;

static const int32_t MAX_ELEMENTS = 1000000;
static const int32_t MAX_VALUE = 10000000;

int main() {
  srand(time(NULL));

  auto startTime = chrono::high_resolution_clock::now();
  vector<int32_t> array(MAX_ELEMENTS);
  for (int32_t at = 0; at < MAX_ELEMENTS; ++at) {
    array[at] = rand() % MAX_VALUE;
  }
  auto endTime = chrono::high_resolution_clock::now();
  auto duration = chrono::duration_cast<chrono::milliseconds>(endTime - startTime).count();
  cout << "Array generated in " << duration << " ms" << endl;

  startTime = chrono::high_resolution_clock::now();
  sort(array.begin(), array.end());
  endTime = chrono::high_resolution_clock::now();
  duration = chrono::duration_cast<chrono::milliseconds>(endTime - startTime).count();
  cout << "Array sorted in " << duration << " ms" << endl;

  return 0;
}